### PR TITLE
Update glossary.md

### PR DESCRIPTION
**Pull request by lkhphuc**
Mình có tổng hợp một số thay đổi trong glossary mọi người thảo luận thêm. Sau khi chốt mình có thể quay lại sửa các chương đã merged theo glossary mới nhât.
- | benchmark                         | chuẩn so sánh   
- | gradient descent                  | hạ dốc                           
- | hand-engineering                  | thiết kế thủ công
- | optimizing metric                 | phép đo cần tối ưu                  
- | satisficing metric                | phép đo cần thỏa mãn           

- fit/overfit/underfit theo mình nên giữ nguyên vì từ fit nó đã bao gồm rất nhiều nghĩa trong rất nhiều hoàn cảnh rồi. Ngay cả native speaker khi gặp những từ này trong ML đều cần được giải thích thì mới hiểu đươc.

- neural network nên giữ nguyên là neural network. tương tự như các từ ML, DL, NN là một cụm viết tắt khá phổ biến rồi, như ANN, RNN, CNN, deep NN. Giữ nguyên NNN sẽ giúp người học dễ nhận ra được các mối quan hệ như trên.

### CODE REVIEW

File: glossary.md [hunk 1]
```diff
@@ -31,29 +31,30 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
...
-| satisficing metric                |                                                                |
+| satisficing metric                 | phép đo cần thỏa mãn                                            |
 | single-number evaluation metric   | phép đo đơn trị                                                |
 | spam email                        | email rác                                                      |
 | supervised learning               | học có giám sát                                                |
 | test set                          | tập kiểm tra                                                   |
 | training set                      | tập huấn luyện                                                 |
 | true negative                     | âm tính thật                                                   |
 | true positive                     | dương tính thật                                                |
-
```
Comment (sonvx) [inline glossary.md:63]: **underfit** mình nghĩ có thể dịch là **vị khớp**.

File: glossary.md [hunk 2]
```diff
@@ -12,17 +12,17 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
 | English                           | Tiếng Việt (Nếu bạn bổ sung thì KHÔNG căn chỉnh các dòng khác) |
 |-----------------------------------|----------------------------------------------------------------|
 | accuracy                          | độ chính xác                                                   |
-| benchmark                         |                                                                |
+| benchmark                         | chuẩn so sánh                                                  |
```
Comment (ngcthuong) [inline glossary.md]: benchmark không hẳn là chuẩn so sánh. Nó giống như là có bảng so sánh các/nhiều thuật toán với nhau với cùng điều kiện, chỉ số đánh giá

File: glossary.md [hunk 3]
```diff
@@ -31,29 +31,30 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
...
 | machine learning                  | machine learning                                               |
 | metric                            | phép đo                                                        |
 | multiple-number evaluation metric | phép đo đa trị                                                 |
 | negative sample/example           | mẫu âm                                                         |
-| neural network                    | mạng neural                                                    |
-| overfit                           |                                                                |
-| optimizing metric                 |                                                                |
+| neural network                    | neural network                                                 |
+| overfit                           | overfit                                                          |
+| optimizing metric                 | phép đo cần tối ưu                                            |
```
Comment (ngcthuong) [inline glossary.md]: phép đo tối ưu nghe hợp hơn, bỏ **cần** đi là đẹp.

File: glossary.md [hunk 4]
```diff
@@ -31,29 +31,30 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
...
 | positive sample/example           | mẫu dương                                                      |
 | precision                         | precision                                                      |
 | recall                            | recall                                                         |
 | regularization                    | regularization                                                 |
 | running time                      | thời gian chạy                                                 |
-| satisficing metric                |                                                                |
+| satisficing metric                 | phép đo cần thỏa mãn                                            |
```
Comment (ngcthuong) [inline glossary.md]: như trên, bỏ Cần đi thì nghe thuận hơn

File: glossary.md (hunk 2 above)
Comment (sonvx) [reply glossary.md]: A vẫn vote là **đánh giá xếp hạng**.

Comment (ngcthuong) [reply glossary.md]: How about *bảng xếp hạng* (cúp C1)

File: glossary.md (hunk 3 above)
Comment (lkhphuc) [reply glossary.md]: Mình ban đầu cũng nghĩ vạy nhưng lại sợ "phep1 đo tối ưu" nhầm lẫn sang "optimal metric".

File: glossary.md (hunk 1 above)
Comment (lkhphuc) [reply glossary.md:63]: nếu dịch underfit thì phải dịch cả "overfit" và "fit" sao cho ngưòi đọc hiểu đuợc sự liên quan của cả ba thuật ngữ trên. Bạn có góp ý gì cho 2 từ kia không?
Nói thiệt là mình cũng không biết "vị khớp" là gì :)).

Comment (ngcthuong) [reply glossary.md:63]: Dịch là "thấp khớp" thì hơi kì cục

**Comment by tiepvupsu** [comment]
Nhận tiện, có nhiều từ đã được bên nhóm dịch Deep Learning bàn rất kỹ [ở đây](https://docs.google.com/spreadsheets/d/1Hcd4AqR7Xzd08Ws0gwhZ-LX6XNzJKgCDy0VcFc3D66c/edit#gid=0), nhóm mình có thể lấy tham khảo:

File: glossary.md [hunk 5]
```diff
@@ -12,17 +12,17 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
...
 | constrain                         | ràng buộc                                                      |
 | development set                   | tập phát triển                                                 |
 | deep learning                     | deep learning                                                  |
 | distribution                      | phân phối                                                      |
-| gradient descent                  |                                                                |
+| gradient descent                  | hạ dốc                                                         |
 | heuristic                         | thực nghiệm                                                    |
 | hidden unit                       | nút ẩn                                                         |
-| hand-engineering                  | thiết kế bằng tay                                              |
+| hand-engineering                  | thiết kế thủ công                                              |
```
Comment (tiepvupsu) [inline glossary.md]: 👍

File: glossary.md [hunk 6]
```diff
@@ -31,29 +31,30 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
 | false negative                    | âm tính giả                                                    |
 | false positive                    | dương tính giả                                                 |
 | feature                           | đặc trưng                                                      |
+| fit                                | fit                                                             |
 | layer                             | tầng                                                           |
 | linear regression                 | hồi quy tuyến tính                                             |
-| learning curve                    |                                                                |
+| learning curve                    | đồ thị học                                                     |
```
Comment (tiepvupsu) [inline glossary.md]: Từ này mình nghĩ dịch là 'quá trình học' là hợp lý hơn. Từ curve không có nghĩa là đồ thị mà có nghĩa là 'quá trình'. Đây là cách giải nghĩa bằng tiếng Anh:
> the rate of a person's progress in gaining experience or new skills.

Ban đầu mình nghĩ tới từ 'tiến trình' vì nó có nghĩa 'quá trình tiến lên' nhưng nghĩ lại sẽ có lúc trong sách từ 'learning curve' không hẳn là luôn 'tiến lên', ví dụ khi overfitting.

**Comment by tiepvupsu** [comment]
Mình vừa cập nhật file [glossary.md](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/glossary.md) bằng cách thêm link thảo luận tới pull request này nên có conflict. Bạn vui lòng resolve lại nhé. Cảm ơn bạn.

File: glossary.md (hunk 2 above)
Comment (lkhphuc) [reply glossary.md]: Mình chuyển thành "đánh giá xếp hạng" theo @sonvx giống như bên [này](https://docs.google.com/spreadsheets/d/1Hcd4AqR7Xzd08Ws0gwhZ-LX6XNzJKgCDy0VcFc3D66c/edit#gid=0) luôn

**Comment by lkhphuc** [comment]
Mình đã resolve conflict hiện tại. Sau khi chốt thuật ngữ nào se fix trong PR này, mình có thể add thêm commit sửa lại toàn bộ những chương đã dịch theo thuật ngữ đó.

**Comment by tiepvupsu** [comment]
Cảm ơn bạn. Mình đồng ý với cách dịch của bạn. Chờ anh @sonvx và @ngcthuong vào bình luận thêm.

Nhân tiện, diff của pull request này có thể nhìn dễ hơn ở dạng [rich text](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/pull/87/files?short_path=f07b738#diff-f07b73835a21496fc01afa719cfd3ba4)

**Comment by ngcthuong** [comment]
> Cảm ơn bạn. Mình đồng ý với cách dịch của bạn. Chờ anh @sonvx và @ngcthuong vào bình luận thêm.
> 
> Nhân tiện, diff của pull request này có thể nhìn dễ hơn ở dạng [rich text](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/pull/87/files?short_path=f07b738#diff-f07b73835a21496fc01afa719cfd3ba4)

Thêm 
- cross validation 
- cross validation set 
- training distribution?
- test distribution? 
- detector

vào nhé? Chương 5 có thuật ngữ đó.

File: glossary.md (hunk 1 above)
Comment (sonvx) [reply glossary.md:63]: Trc đây bên DLBook cũng consider các cách dịch **vị khớp, kém khớp, không khớp** và rồi **vị khớp** đc chọn. Trong đó **kém khớp** cũng ko khác **thấp khớp** và `thấp khớp` thì kì cục thật :D.

Comment (sonvx) [reply glossary.md:63]: > nếu dịch underfit thì phải dịch cả "overfit" và "fit" sao cho ngưòi đọc hiểu đuợc sự liên quan của cả ba thuật ngữ trên. Bạn có góp ý gì cho 2 từ kia không?
> Nói thiệt là mình cũng không biết "vị khớp" là gì :)).

**Overfit** là **quá khớp** bạn nhé.

**Comment by patrickphat** [comment]
Em có học qua 5 khóa của bác Andrew thì em sure bác Andrew dùng 2 từ này khá nhiều:
Iteration: tiến trình
Iterate: tiến triển
Hai từ này được dùng khác nghĩa với chữ "vòng lặp" dù cũng có nghĩa liên quan sâu xa nào đó.

**Comment by ngcthuong** [comment]
> Em có học qua 5 khóa của bác Andrew thì em sure bác Andrew dùng 2 từ này khá nhiều:
> Iteration: tiến trình
> Iterate: tiến triển
> Hai từ này được dùng khác nghĩa với chữ "vòng lặp" dù cũng có nghĩa liên quan sâu xa nào đó.

Iteration là thuật ngữ vòng lặp trong viết code. Cần đặt trong ngữ cảnh để phân biệt

**Comment by tiepvupsu** [comment]
> > Cảm ơn bạn. Mình đồng ý với cách dịch của bạn. Chờ anh @sonvx và @ngcthuong vào bình luận thêm.
> > Nhân tiện, diff của pull request này có thể nhìn dễ hơn ở dạng [rich text](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/pull/87/files?short_path=f07b738#diff-f07b73835a21496fc01afa719cfd3ba4)
> 
> Thêm
> 
> * cross validation
> * cross validation set
> * training distribution?
> * test distribution?
> * detector
> 
> vào nhé? Chương 5 có thuật ngữ đó.

Thêm riêng vào một PR khác để approve và merge cái này đã cậu.

Không nên gom quá nhiều thứ vào một PR.

**Review by tiepvupsu — APPROVED** [review]

**Comment by tiepvupsu** [comment]
> Mình đã resolve conflict hiện tại. Sau khi chốt thuật ngữ nào se fix trong PR này, mình có thể add thêm commit sửa lại toàn bộ những chương đã dịch theo thuật ngữ đó.

Mình đã approve lỡ merge. Xin lỗi bạn, bạn thêm một PR khác để sửa lại các chương đã dịch nhé.

Cảm ơn bạn.

File: glossary.md [hunk 7]
```diff
@@ -13,20 +13,21 @@ Nếu một từ chưa có trong bảng thuật ngữ dưới đây, các bạn
 
 ## Anh - Việt
 
+
 | English                           | Tiếng Việt (Nếu bạn bổ sung thì KHÔNG căn chỉnh các dòng khác) | Thảo luận tại                |
 |-----------------------------------|----------------------------------------------------------------|------------------------------|
 | accuracy                          | độ chính xác                                                   |                              |
-| benchmark                         |                                                                | [#87](http://bit.ly/2BvfPYA) |
+| benchmark                         | đánh giá xếp hạng                                              | [#87](http://bit.ly/2BvfPYA) |
```
Comment (thanhnd0101) [inline glossary.md:20]: - benchmark theo em hiểu, nó là biểu đồ được tính toán đo lường để biểu diễn (vẽ) ra theo một độ đo nào đó bằng biểu đồ (thường là biểu đồ đường: độ đo / thời gian) để người ta có thể biểu thị được chất lượng hoạt động của một tính năng nào đó. Còn dùng cho để đánh giá, xếp hạng hay thống kê gì thì tùy vào mục đích của người vẽ nên. Nên em nghĩ là dịch là `đánh giá xếp hạng` không được chính xác lắm. Em nghĩ nên để tiếng anh luôn.

Comment (ngcthuong) [reply glossary.md:20]: Benchmark không phải là biểu đồ tính toán. Biểu đồ tính toán chỉ là một cách họ thể hiện benchmark - đánh giá xếp hạng thôi em. 

Ví dụ em có thể tìm kiếm các bài báo, công trình nghiên cứu có chữ "benchmark", mục tiêu cuối cùng là xếp hạng các thuật toán theo các chỉ số, điều kiện test tương ứng.

File: glossary.md (hunk 1 above)
Comment (ngcthuong) [reply glossary.md:63]: Những từ hợp với nghĩa underfit

hạ khớp (nhưng hạ cũng là động từ)
thiếu khớp ( giống như thiếu niên)
dưới khớp (như ở phía dưới, thấp hơn)
vị khớp (trong vị thành niên)
Nghe qua thì thấy từ "thiếu" với "vị" có vẻ ok. Từ vị thì nghe nó lạ vì mọi người ít dùng hơn từ thiếu.

Còn anh em ngồi nói chuyện vui với nhau thì nên dùng từ chuẩn là "thấp khớp"

**Comment by Khiem179** [comment]
Mình nghĩ là từ hạ gradient nên dịch thành hạ đạo hàm thì sẽ gần gũi hơn. Với cả từ "exploding gradient" nên dịch thành 'phân kỳ đạo hàm'

File: glossary.md (hunk 1 above)
Comment (tien238lnd) [reply glossary.md:63]: Có ai nghĩ đến từ **chưa khớp** không nhỉ? Các bạn thấy từ này thế nào?
